### PR TITLE
bugfix: support jdk9+ compile code

### LINKED ISSUE #1410: Seata Support jdk > 1.8
- [x] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.
在jdk >1.8 的版本下，seata无法正常运行

现象1:
使用jdk1.11编译seata-all,启动运行client抛出NoSuchMethodError异常
```
java.lang.NoSuchMethodError: java.nio.ByteBuffer.flip()Ljava/nio/ByteBuffer;
```
server端抛出OutOfMemoryError和TooLongFrameException异常
```
java.lang.OutOfMemoryError: Direct buffer memory
io.netty.handler.codec.TooLongFrameException: Adjusted frame length exceeds 8388608: 16777216 - discarded
```
现象2:
seata-server在jdk >1.8 的环境下启动记得也是报TooLongFrameException异常

### Environment:

- JDK version : > 1.8
- OS : Mac OX、Ubuntu
- Others:

### PR DESCRIPTION
**Pull request by funky-eyes**
- [x] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅰ. Describe what this PR did
使用jdk9及以上版本编译seata源码后在jdk8中进行引入,再默认序列化下,会出现
java.lang.NoSuchMethodError: java.nio.ByteBuffer.flip()Ljava/nio/ByteBuffer)
需要将ByteBuffer转换为Buffer再调用相关函数

### Ⅱ. Does this pull request fix one issue?

fixes #1410 #2223

### CODE REVIEW

**Review by slievrly — COMMENTED** [review]
No problem on implementation function, but why not consider using instance classes to wrap it.

**Comment by a364176773** [comment]
> No problem on implementation function, but why not consider using instance classes to wrap it.

包装类改动的代码范围会比较广，工具类相对比较通用
wrapper class changes the scope of the code will be broader, tool class is relatively general

**Comment by a364176773** [comment]
@slievrly @wangliang181230 PTAL

**Review by wangliang181230 — APPROVED** [review]
LGTM

File: common/src/main/java/io/seata/common/util/BufferUtils.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+ *
+ * @author funkye
+ */
+public class BufferUtils {
+
+    /**
+     * @param buffer byteBuffer
+     */
+    public static void flip(Buffer buffer) {
+        buffer.flip();
```
Comment (wangliang181230) [inline common/src/main/java/io/seata/common/util/BufferUtils.java:48]: 返回类型，是不是应该return一下？没准其他哪里会用到。
下面的方法也看一下吧。

**Pull request merged**